frontend/packages/pipelines-plugin/src/components/pac/PacAppForm.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import { FormGroup, TextInputTypes } from '@patternfly/react-core';
+import { FormikProps, FormikValues } from 'formik';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import FormSection from '@console/dev-console/src/components/import/section/FormSection';
+import { k8sListResourceItems } from '@console/dynamic-plugin-sdk/src/utils/k8s';
+import { ExternalLink } from '@console/internal/components/utils';
+import { RouteModel } from '@console/internal/models';
+import { RouteKind } from '@console/internal/module/k8s';
+import { FormFooter, FormBody, InputField, FlexForm } from '@console/shared';
+import { GITHUB_APP_DOC, PIPELINES_NS } from '../../const';
+import PacPermissions from './PacPermissions';
+
+const PacAppForm: React.FC<FormikProps<FormikValues> & {
+  formRef: React.LegacyRef<HTMLFormElement>;
+}> = ({ errors, handleReset, handleSubmit, status, isSubmitting, values, formRef }) => {
+  const { t } = useTranslation();
+  const [manifestVal, setManifestVal] = React.useState('');
+  const [elRoute, setElRoute] = React.useState<string>();
+
+  React.useEffect(() => {
```

Comment:
Shall we extract this to a separate file, hooks-github-manifest ? It will be easy to add test and maintain  it (if any changes required in future)